--- Tesserae/src/Components/OmniBox.cs
+++ Tesserae/src/Components/OmniBox.cs
@@ -306,6 +306,21 @@
                             }
                         }
                     }
+                    else if (ke.key == "ArrowDown")
+                    {
+                        if (_hideSuggestions != null && _currentSuggestionButtons.Count > 0)
+                        {
+                            e.preventDefault();
+                            UpdateHighlightedSuggestion(1);
+                        }
+                    }
+                    else if (ke.key == "ArrowUp")
+                    {
+                        if (_hideSuggestions != null && _currentSuggestionButtons.Count > 0)
+                        {
+                            e.preventDefault();
+                            UpdateHighlightedSuggestion(-1);
+                        }
+                    }
                 });
                 _searchInput.addEventListener("keyup", (e) => KeyUp?.Invoke(this, e.As<KeyboardEvent>()));
                 _searchInput.addEventListener("keypress", (e) => KeyPress?.Invoke(this, e.As<KeyboardEvent>()));
@@ -1187,6 +1202,21 @@

             }, 300);
         }
+
+        private void UpdateHighlightedSuggestion(int offset)
+        {
+            if (_currentSuggestionButtons.Count == 0) return;
+
+            if (_highlightedSuggestionIndex >= 0 && _highlightedSuggestionIndex < _currentSuggestionButtons.Count)
+            {
+                _currentSuggestionButtons[_highlightedSuggestionIndex].Render().classList.remove("tss-omnibox-suggestion-highlight");
+                _currentSuggestionButtons[_highlightedSuggestionIndex].Render().style.background = "";
+            }
+
+            _highlightedSuggestionIndex = (_highlightedSuggestionIndex + offset + _currentSuggestionButtons.Count) % _currentSuggestionButtons.Count;
+
+            _currentSuggestionButtons[_highlightedSuggestionIndex].Render().classList.add("tss-omnibox-suggestion-highlight");
+            _currentSuggestionButtons[_highlightedSuggestionIndex].Render().style.background = "var(--tss-hover-background-color)";
+        }

         public OmniBox SetSearchRightText(string text)
